File: verify_cost_sheet.ts
import { test, expect } from '@playwright/test';

test('verify cost sheet changes', async ({ page }) => {
  await page.goto('http://localhost:3000/auth/login');
  await page.fill('input[name="email"]', 'admin@demo.com');
  await page.fill('input[name="password"]', 'demo123');
  await page.click('button[type="submit"]');
  await page.waitForURL('http://localhost:3000/dashboard');

  // Scroll the sidebar to find COSTOS
  const sidebar = page.locator('aside');
  await sidebar.hover();
  await page.mouse.wheel(0, 500);

  // Find and click COSTOS
  const costosLink = page.getByRole('link', { name: /COSTOS/i });
  await costosLink.scrollIntoViewIfNeeded();
  await costosLink.click();

  await page.waitForURL(**/terminal/cost-sheet**);
  await page.waitForSelector('text=FICHA DE COSTOS');

  // Check Annex II
  await page.click('button:has-text("Anexos")');
  await page.click('button:has-text("Anexo II")');

  // Verify Description column in Annex II doesn't have datalist
  const annexIIDescription = page.locator('table >> tr >> td >> input').first();
  const listAttr = await annexIIDescription.getAttribute('list');
  console.log('Annex II Description list attribute:', listAttr);

  // Take screenshot of Annex II
  await page.screenshot({ path: '/home/jules/verification/annex_ii.png', fullPage: true });

  // Check Annex IV suggestions
  await page.click('button:has-text("Anexo IV")');
  await page.screenshot({ path: '/home/jules/verification/annex_iv.png', fullPage: true });

  // Go back to Main Table
  await page.click('button:has-text("Tabla Principal")');
  await page.screenshot({ path: '/home/jules/verification/main_table.png', fullPage: true });
});

File: verification/verify_cost_sheet.py
from playwright.sync_api import Page, expect, sync_playwright
import time

def test_cost_sheet_verification(page: Page):
    # Go to cost sheets page
    page.goto("http://localhost:3000/cost-sheets")

    # Wait for initial load
    time.sleep(5)

    # Click 'Ejemplo'
    # Try multiple ways to find it
    try:
        page.get_by_text("Ejemplo").first.click()
    except:
        page.click("button:has-text('Ejemplo')")

    time.sleep(2)

    # Select a section group to show the table
    try:
        page.get_by_text("SECCIONES 1 - 3").first.click()
    except:
        page.get_by_text("Sección").first.click()

    time.sleep(2)

    # Take screenshot of the table
    page.screenshot(path="/app/verification/cost_sheet_table.png", full_page=True)
    print("Screenshot saved to /app/verification/cost_sheet_table.png")

if __name__ == "__main__":
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        try:
            test_cost_sheet_verification(page)
        finally:
            browser.close()
